File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        compMap = dict()
        for i, num in enumerate(nums):
            comp = target - num
            if num in compMap:
                return [compMap[num],i]
            compMap[comp] = i

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    
    var hashmap = map[int]int {};

    for i := 0; i < len(nums); i++ {
        value, ok := hashmap[target - nums[i]];
        
        if ok {
            return []int {value, i};
        }
        
        hashmap[nums[i]] = i;
    }
    return []int {};
}
